### PR TITLE
Additional checks that prevent a page from re-rendering unnecessarily

### LINKED ISSUE #253: Don't redownload things if we already have them.
Relevant especially to survey questions page.

### PR DESCRIPTION
**Pull request by ttavenner**
Prevents changes to state in `componentWillMount` from triggering multiple unnecessary re-renders.

Fix #253

### CODE REVIEW

**Comment by wbprice** [comment]
This seems fine.  Have you found that a new record is created when the user visits the results page too?

**Comment by ttavenner** [comment]
@wbprice No new records were created on the results page but it was calling the match service multiple times for the same reason as the other pages. I cleaned this up.

File: frontend/js/components/environments/Results.js [hunk 1]
```diff
@@ -31,7 +31,9 @@ class Results extends Component {
     }
 
     componentWillMount() {
```
Comment (wbprice) [inline frontend/js/components/environments/Results.js:33]: `componentWillMount` is called on the server and in the client.  `componentDidMount` is only called in the client.  Did you mean to put this in that hook?

File: frontend/js/components/environments/Survey.js [hunk 2]
```diff
@@ -39,8 +39,10 @@ class Survey extends Component {
             fetchSurveyQuestions(SurveyId)
         );
 
-        if (this.props.location.query.newSurvey) {
-            return this.props.dispatch(createSurveyResult(SurveyId, fetchQuestions));
+        if (!this.props.survey.SurveyResultId) {
```
Comment (wbprice) [inline frontend/js/components/environments/Survey.js:42]: The logic for this should be:
```
If the user asks for a new survey OR a survey result is not defined THEN
do the stuff
```

**Review by wbprice — CHANGES_REQUESTED** [review]
Pull in these fixes to prevent a reference error when rendering on the server. https://github.com/ttavenner/okcandidate-platform/pull/2

**Review by wbprice — APPROVED** [review]

**Pull request merged**